Add modal to new home page

cryptofox17 opened the pull request:
# Related Issue
Fixes: #207 

# Prosposed Changes
- added modal to new home page

# Screenshots
![Screenshot (150)](https://user-images.githubusercontent.com/64403591/115145173-db88ff00-a06d-11eb-941c-fe9e856a447e.png)
![Screenshot (149)](https://user-images.githubusercontent.com/64403591/115145174-dcba2c00-a06d-11eb-9f88-f18c41bf078d.png)

cryptofox17 commented:
@garg3133 made a pr for this already but there were too many conflicts so i decided to make a new one.

garg3133 commented:
@cryptofox17 Sorry for telling you this late, but can you use the Bootstrap modals for this issue? We are using bootstrap on our new home page, which we were not using in the older home page, that's why we needed to implement everything ourselves. But as we are using Bootstrap on this page, implementing the Donate Us modal using bootstrap would be a lot cleaner and easier to read.

garg3133 commented:
Other than that, the design looks good. Just a little space is required above that OR line and you can avoid using that circle around the close button, which looks a bit odd.

cryptofox17 commented:
@garg3133  I have removed most of the custom css and added required bootstrap classes instead 
![Screenshot (152)](https://user-images.githubusercontent.com/64403591/115148815-f021c300-a07e-11eb-8225-09b695ab302e.png)
![Screenshot (151)](https://user-images.githubusercontent.com/64403591/115148817-f1eb8680-a07e-11eb-9972-b55097b26d04.png)

garg3133 commented:
@cryptofox17 I not only meant using bootstrap classes for stuff line margins, centre-aligning, etc., but I also meant using Bootstrap for modals itself.

See https://www.w3schools.com/bootstrap4/bootstrap_modal.asp

cryptofox17 commented:
@garg3133 done 

![Screenshot (157)](https://user-images.githubusercontent.com/64403591/115414791-fc8b5480-a213-11eb-9fb5-eeeae1ce2859.png)
![Screenshot (156)](https://user-images.githubusercontent.com/64403591/115414792-fe551800-a213-11eb-99c6-7acf2eb097a8.png)

garg3133 commented:
@cryptofox17 I am currently busy with my college project and assignment submissions followed by my end-sem exams, so I might not be available for a few days to review your PR. Sorry for the delay.

cryptofox17 commented:
No problem, good luck with the exams

tushhr reviewed (changes requested):
Hey @cryptofox17, sorry for such a late review. I have reviewed your PR and it was almost perfect! I have posted a few suggestions kindly check.

Also  @cryptofox17 there is a major bug in your PR, whenever I open the modal and close it, it literally adds the padding in the body, I am attaching the screenshot for reference

![image](https://user-images.githubusercontent.com/63875632/117249371-10af9280-ae5f-11eb-98ed-ac6f171a7f1f.png)

Can you see that right padding? (Image is after 5-6 attempts of opening and closing the Modal)
Actually, it's the issue of bootstrap itself, and they have one PR open for it, but don't know when they will merge it, till then you can fix it manually.

tushhr commented:
Also one thing I forgot to add, the color of the `Donate` in the navbar is "White", and in your code, it changes to black, can you fix it too?

cryptofox17 commented:
@tushhr done, let me know if its fine now

![Screenshot (181)](https://user-images.githubusercontent.com/64403591/117352574-c2d17380-aecc-11eb-9b11-4caa72167f80.png)

tushhr commented:
Awesome @cryptofox17!
Sorry but I don't have the rights to merge any PR, so @garg3133 will review it one final time and merge it soon. Meanwhile, you can take up other issues.

Sorry for the inconvenience and thanks for your contribution🎉
